kevlar/call.py

```diff
@@ -161,79 +170,128 @@ def call_snv(target, query, offset, length, ksize):
+        offset *= -1
+        targetshort = True
+        alt, refr, altwindow, refrwindow = insertion_allele(
+            query, target, offset, ksize, leftmatch, indellength
+        )
+    else:
+        targetshort = False
+        refr, alt, refrwindow, altwindow = deletion_allele(
+            target, query, offset, ksize, leftmatch, indellength
+        )
+    # This assertion is no longer valid when query is longer than target
+    # assert len(refr) == indellength + 1
+    localcoord = leftmatch
+    if not targetshort:
+        localcoord += offset
+    seqid, globalcoord = local_to_global(localcoord, target.name)
+    return [VariantIndel(seqid, globalcoord - 1, refr, alt, VW=altwindow,
+                         RW=refrwindow, IK=str(len(query.ikmers)))]
+
+
+def call_insertion(target, query, offset, ksize, leftmatch, indellength):
+    if offset < 0:
+        offset *= -1
+        targetshort = True
+        alt, refr, altwindow, refrwindow = deletion_allele(
+            query, target, offset, ksize, leftmatch, indellength
+        )
+    else:
+        targetshort = False
+        refr, alt, refrwindow, altwindow = insertion_allele(
+            target, query, offset, ksize, leftmatch, indellength
+        )
+
+    assert len(alt) == indellength + 1
+    localcoord = leftmatch
+    if not targetshort:
+        localcoord += offset
     seqid, globalcoord = local_to_global(localcoord, target.name)
-    return [VariantIndel(seqid, globalcoord - 1, refr, insertion, VW=window,
+    return [VariantIndel(seqid, globalcoord - 1, refr, alt, VW=altwindow,
                          RW=refrwindow, IK=str(len(query.ikmers)))]
 
 
 def make_call(target, query, cigar, ksize):
-    snvmatch = re.search('^(\d+)D(\d+)M(\d+)D$', cigar)
-    snvmatch2 = re.search('^(\d+)D(\d+)M(\d+)D(\d+)M$', cigar)
+    snvmatch = re.search('^(\d+)([DI])(\d+)M(\d+)[DI]$', cigar)
+    snvmatch2 = re.search('^(\d+)([DI])(\d+)M(\d+)[DI](\d+)M$', cigar)
     if snvmatch:
         offset = int(snvmatch.group(1))
-        length = int(snvmatch.group(2))
+        if snvmatch.group(2) == 'I':
+            offset *= -1
+        length = int(snvmatch.group(3))
         return call_snv(target, query, offset, length, ksize)
-    elif snvmatch2 and int(snvmatch2.group(4)) <= 5:
+    elif snvmatch2 and int(snvmatch2.group(5)) <= 5:
         offset = int(snvmatch2.group(1))
-        length = int(snvmatch2.group(2))
+        if snvmatch2.group(2) == 'I':
+            offset *= -1
+        length = int(snvmatch2.group(3))
         return call_snv(target, query, offset, length, ksize)
 
-    indelmatch = re.search('^(\d+)D(\d+)M(\d+)([ID])(\d+)M(\d+)D$', cigar)
-    indelmatch2 = re.search('^(\d+)D(\d+)M(\d+)([ID])(\d+)M(\d+)D(\d+)M$',
-                            cigar)
+    indelmatch = re.search(
+        '^(\d+)([DI])(\d+)M(\d+)([ID])(\d+)M(\d+)[DI]$', cigar
+    )
+    indelmatch2 = re.search(
+        '^(\d+)([DI])(\d+)M(\d+)([ID])(\d+)M(\d+)[DI](\d+)M$', cigar
+    )
```

Comment:
This is the critical change to the PR with respect to INDELs. Most other changes in this file were to compensate for this change.